microstream/src/test/groovy/io/micronaut/microstream/health/MicrostreamHealthIndicatorFuncSpec.groovy

```diff
@@ -0,0 +1,81 @@
+package io.micronaut.microstream.health
+
+import io.micronaut.context.BeanContext
+import io.micronaut.health.HealthStatus
+import io.micronaut.management.health.indicator.HealthResult
+import io.micronaut.test.annotation.MockBean
+import io.micronaut.test.extensions.spock.annotation.MicronautTest
+import jakarta.inject.Inject
+import jakarta.inject.Named
+import one.microstream.storage.embedded.types.EmbeddedStorageManager
+import org.reactivestreams.Subscriber
+import org.reactivestreams.Subscription
+import spock.lang.Specification
+import spock.util.concurrent.PollingConditions
+
+import java.util.function.Consumer
+
+@MicronautTest(startApplication = false)
+class MicrostreamHealthIndicatorFuncSpec extends Specification {
+
+    @Inject
+    BeanContext beanContext
+
+    EmbeddedStorageManager mockStorageManager = Mock()
+
+    def "#desc manager is #expectedStatus"() {
+        given:
+        def healthIndicator = beanContext.getBean(MicrostreamHealthIndicator)
+        List<HealthResult> results = []
+
+        when:
+        healthIndicator.result.subscribe(new TestSubscriber<HealthResult>({ HealthResult t -> results << t }))
+
+        then:
+        new PollingConditions().eventually {
+            results.status == [expectedStatus]
+        }
+
+        and:
+        _ * mockStorageManager.isRunning() >> isRunning
+
+        where:
+        expectedStatus    | isRunning | desc
+        HealthStatus.DOWN | false     | 'not running'
+        HealthStatus.UP   | true      | 'running'
+    }
+
+    @MockBean(bean = EmbeddedStorageManager, named = "mock-manager")
+    @Named("mock-manager")
+    EmbeddedStorageManager getEmbeddedStorageManager() {
+        mockStorageManager
+    }
+
+    private class TestSubscriber<T> implements Subscriber<T> {
```

Comment:
this could be simpler using project-reactor-test

Reply:
resolved via https://github.com/micronaut-projects/micronaut-microstream/pull/22/commits/13cd774baa6e4f5425d38936e254723a1be0a917